### DISCUSSION TITLE
Can't find `vineyardd` executable after `pip install vineyard`

### DISCUSSION THREAD

**Discussion by Caviarion** [v6d-io/v6d#186] [Q&A] [answered]
I'm installing inside `virtualenv` with `python 3.8.3`

Version of `vineyard`: 0.1.8

After successful installation, I can't find `vineyard` executable anywhere

Any Idea?

**Comment by sighingnow** [comment]
The vineyardd binary hasn't been included into the wheel package yet. The prebuilt vineyardd binary for linux could be found at https://github.com/alibaba/libvineyard/releases/tag/v0.1.8.

This issue will be fixed in vineyard v0.1.9, before the end of this week.

**Comment by sighingnow** [comment, marked as answer]
We already have an issue for that: https://github.com/alibaba/libvineyard/issues/165.

**Reply by sighingnow** [reply]
v0.1.9 released: https://pypi.org/project/vineyard/0.1.9/

**Comment by Caviarion** [comment]
I tried to use the prebuilt binary.
After successfully starting the server with `sudo ./vineyard`, I got

```
$ sudo ./vineyardd                                                          
I0202 17:58:16.243157 247018 vineyardd.cc:64] Hello vineyard!
I0202 17:58:16.248601 247018 meta_service.h:130] start!
I0202 17:58:16.249924 247018 meta_service.h:702] Instance join: 0
I0202 17:58:16.343680 247018 meta_service.h:373] Decide to set rank as 1
I0202 17:58:16.343721 247018 meta_service.h:702] Instance join: 1
I0202 17:58:16.346527 247018 meta_service.h:383] start background etcd watch, since 7
I0202 17:58:16.346827 247018 ipc_server.cc:43] Vineyard will listen on "/var/run/vineyard.sock" for IPC
I0202 17:58:16.346879 247018 rpc_server.cc:50] Vineyard will listen on 0.0.0.0:9600 for RPC
```

And `vineyard.sock` appears at `/var/run` with perm `srwxr-xr-x  1 root  root       0 Feb  2 17:58 vineyard.sock=`

Everything seems good but executing `vineyard.connect` gives me

```
Traceback (most recent call last):
  File "test.py", line 2, in <module>
    client = vineyard.connect('/var/run/vineyard.sock')
vineyard._C.ConnectionFailedException: Failed to connect to vineyardd: Failed to resolve IPC socket or RPC endpoint of vineyard server
```

FYI, I'm using python package installed from pip and `vineyard` binary downloaded from above. Could this be the reason ?

**Reply by sighingnow** [reply]
Hi @Caviarion,

Vineyardd is running using the root directory but the python script not, thus it doens't a write permission bit for the UNIX domain socket.

There're two options to fix that:

1. launching vineyardd using a non-root user. You could specify the location of UNIX domain socket by the `--socket ...` parameter.
2. running the python script using the same user with vineyardd, i.e., `root` in your case.

Hope that is helpful for you!

**Reply by Caviarion** [reply]
Thx, it works now.

**Reply by sighingnow** [reply]
Hi @Caviarion,

We have just released v0.1.9 with vineyardd bundled in the pypi packages: https://pypi.org/project/vineyard/0.1.9/.
